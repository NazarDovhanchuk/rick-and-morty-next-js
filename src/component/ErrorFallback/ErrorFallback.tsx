/* eslint-disable @typescript-eslint/no-explicit-any */
/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
import React from 'react';

const ErrorFallback = ({ error }: any): JSX.Element => (
  <div role="alert">
    <p>Something went wrong:</p>
    <pre style={{ color: 'red' }}>{error.message}</pre>
  </div>
);

export default ErrorFallback;
