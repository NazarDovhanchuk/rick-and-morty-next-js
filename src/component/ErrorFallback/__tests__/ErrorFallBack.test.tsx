import React from "react";
import rendered from "react-test-renderer";
import { render, cleanup } from "@testing-library/react";

import ErrorFallback from "../ErrorFallback";

describe('Error Components', () => {
  it('it Error Components render', () => {
    const errorComponent = rendered.create(<ErrorFallback error="message" />);
    const tree = errorComponent.toJSON();

    expect(tree).toMatchSnapshot();
  })
})