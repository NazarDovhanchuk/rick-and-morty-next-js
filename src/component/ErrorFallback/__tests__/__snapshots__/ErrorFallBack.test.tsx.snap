// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error Components it Error Components render 1`] = `
<div
  role="alert"
>
  <p>
    Something went wrong:
  </p>
  <pre
    style={
      Object {
        "color": "red",
      }
    }
  />
</div>
`;
